app/kubernetes/horizontal-pod-auto-scaler/converter.js

```diff
@@ -20,4 +23,111 @@ export class KubernetesHorizontalPodAutoScalerConverter {
     res.Yaml = yaml ? yaml.data : '';
     return res;
   }
+
+  static createPayload(data) {
+    const payload = new KubernetesHorizontalPodAutoScalerCreatePayload();
+    payload.metadata.namespace = data.Namespace;
+    payload.metadata.name = data.TargetEntity.Name;
+    payload.spec.minReplicas = data.MinReplicas;
+    payload.spec.maxReplicas = data.MaxReplicas;
+    payload.spec.targetCPUUtilizationPercentage = data.TargetCPUUtilization;
+    payload.spec.scaleTargetRef.apiVersion = data.TargetEntity.ApiVersion;
```

Comment:
I had to change this line to the following during my testing:

```
payload.spec.scaleTargetRef.apiVersion = 'apps/v1';
```

To force the use of `apps/v1` otherwise the HPA would not find the associated Deployment. Can you investigate why?

You can investigate the status of the HPA through the YAML of the application in the application details or via `kubectl describe hpa/<name_of_the_app>`.

Reply:
Actually we are supposed to only leverage `apps/v1` objects (deployments / daemonsets / statefulsets / services), if it's not the case we may have something broken somewhere else in the app.